tests/unit/amo/api/test_users.js

```diff
@@ -0,0 +1,77 @@
+import * as api from 'core/api';
+import { currentUserAccount, userAccount } from 'amo/api/users';
+import {
+  createApiResponse,
+  createUserAccountResponse,
+} from 'tests/unit/helpers';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+
+
+describe(__filename, () => {
+  let mockApi;
+
+  beforeEach(() => {
+    mockApi = sinon.mock(api);
+  });
+
+  describe('currentUserAccount', () => {
+    const mockResponse = () => createApiResponse({
+      jsonData: createUserAccountResponse(),
+    });
+
+    it('fetches the current user profile', () => {
+      const state = dispatchClientMetadata().store.getState();
+
+      mockApi.expects('callApi')
+        .withArgs({
+          auth: true,
+          endpoint: 'accounts/profile',
+          state: state.api,
+        })
+        .returns(mockResponse());
+
+      return currentUserAccount({ api: state.api })
+        .then(() => mockApi.verify());
+    });
+
+    it('throws an error if api state is missing', () => {
+      expect(() => {
+        currentUserAccount({});
+      }).toThrowError(/api state is required/);
+    });
+  });
+
+  describe('userAccount', () => {
+    const mockResponse = () => createApiResponse({
+      jsonData: createUserAccountResponse(),
+    });
+
+    it('fetches the current user profile', () => {
+      const state = dispatchClientMetadata().store.getState();
+      const username = 'tofumatt';
+
+      mockApi.expects('callApi')
+        .withArgs({
+          auth: true,
+          endpoint: `accounts/account/${username}`,
+          state: state.api,
+        })
+        .returns(mockResponse());
+
+      return userAccount({ api: state.api, username })
+        .then(() => mockApi.verify());
+    });
+
+    it('throws an error if api state is missing', () => {
+      expect(() => {
+        userAccount({ username: 'sweet-little-duck' });
```

Comment:
I find these kind of tests are a lot easier to maintain if you use a common helper for the params:

```js
const params = getParams();
delete params.api;
expect(() => userAccount(params)).toThrowError(...);
```